sky/execution.py

```diff
@@ -167,6 +170,16 @@ def _execute(
             raise ValueError(
                 'Spot recovery is specified in the task. To launch the '
                 'managed spot job, please use: sky spot launch')
+    if task.use_spot and not _is_launched_by_spot_controller:
+        yellow = colorama.Fore.YELLOW
+        bold = colorama.Style.BRIGHT
+        reset = colorama.Style.RESET_ALL
+        logger.info(
+            f'{yellow}Launching an unmanaged spot task, which does not '
+            f'automatically recover from preemptions.{reset}\n{yellow}To get '
+            'automatic recovery, use managed spot instead: '
+            f'{reset}{bold}sky spot launch{reset} {yellow}or{reset} '
+            f'{bold}sky.spot_launch(){reset}.')
```

Comment:
Actually, how about we place the logging in the `backend.provision`?
Reason:
1. We may only want to print the message once, when it is provisioning. In the `backend.provision`, we will have the information, including the actually launched resources and whether the cluster is firstly launched. The current logging will be printed even when user `sky exec mycluster task.yaml`
2. The spot property sounds more tied with the backend, and it would be better to handle it in the backend?

Reply:
It looks like `task.spot_task` is not serialized into / deserialized from a yaml file, which `controller.py` depends on. So when the controller process for a spot job launches and runs `sky.launch(actual_task)`, `actual_task.spot_task` doesn't exist.

On moving it into `backend.provision()`: it needs a `_is_launched_by_spot_controller` to be plumbed through from this function as well.

Given the above, how about we keep the logging in this func? We can add an additional check `Stage.PROVISION in stages` to print it for `sky launch` only.

Reply:
Ahh, that makes sense. I misunderstood the purpose of this. Having the check under the `PROVISION` stage should be fine, though it may still log out the information when the user `sky launch -c existing-cluster` again.

Could not think of a better way at the moment, but I wouldn't say I like the internal argument in the API. It might be better to make it a global config or have a wrapper inner function `_launch` which hash the argument, but not for the `launch` function?

Reply:
We could change this 

https://github.com/skypilot-org/skypilot/blob/master/sky/spot/recovery_strategy.py#L155

into a `execution.internal_launch_from_spot_controller(...)`? That said, it may bloat the interface since the new function will have only one argument difference from the original. 

Or, we can use a `**kwargs` to pass in this bool.

Examples with internal args aren't uncommon:
- https://docs.ray.io/en/latest/ray-core/package-ref.html#placement-group
- https://docs.ray.io/en/latest/ray-core/package-ref.html#ray-shutdown
- https://docs.ray.io/en/latest/ray-core/package-ref.html#ray-put

Reply:
Good point! I bloating the interface seems worse to me. Let's keep the internal argument then. : )

Reply:
Moved logging, and a second `launch` would not trigger the logging again.